### DISCUSSION TITLE
Forwarding on wrong Form

### DISCUSSION THREAD

**Discussion by Ultraxime** [rwf2/Rocket#2794] [Questions]
Hi,

I'm trying to implement a form that is to be filled progressively.
My idea is to use different structs with more and more fields and forward if fields are missing for example: 
```rust
#[derive(Deserialize, FromForm)]
pub struct Level {
    level: u8,
}

#[post("/", rank = 2, data="<input>")]
pub fn level_post(input: rocket::form::Form<Level>) -> Template {
  // Do something
}

#[derive(Deserialize, FromForm)]
pub struct Field<'r> {
    field: &'r str,
    level: u8,
}

#[post("/", rank = 1, data="<input>")]
pub fn field_post(input: rocket::form::Form<Field>) -> Template {
    // Do something
}
```

But it fails on the first route (`field_post`) because the form is missing a field and returns an error 422 instead of forwarding to `level_post`.

Did I miss something or it's not something that can be done?

Thank you for any help.

**Comment by the10thWiz** [comment]
Depending on how complex your senario is, you may be able to just use `field: Option<&'r str>` to allow it to be missing.

Otherwise, if you have control over the names of the fields, you could use something like:
```rust
#[derive(Debug, FromForm)]
pub struct Fields<'r> {
    field: &'r str,
}

#[derive(Debug, FromForm)]
pub struct Levels {
    level: u8,
}

#[derive(Debug, FromForm)]
pub struct Forms<'r> {
    fields: Option<Fields<'r>>,
    levels: Option<Levels>,
}
```

The level field would now be `levels.level` in the HTML, but this allows grouping multiple fields together. When combined with strict parsing, this will require the user to send either none or all of the feilds in a substruct.
